src/main/java/org/folio/circulation/domain/ConfigurationRepository.java

```diff
@@ -26,20 +26,21 @@ public ConfigurationRepository(Clients clients) {
     configurationClient = clients.configurationStorageClient();
   }
 
-  public CompletableFuture<Result<LoanAndRelatedRecords>> lookupTimeZone(
-    LoanAndRelatedRecords relatedRecords) {
+  public CompletableFuture<Result<LoanAndRelatedRecords>> lookupTimeZoneForLoanRelatedRecords(
+    Result<LoanAndRelatedRecords> relatedRecords) {
```

Comment:
Can we just return the time zone here without passing in the LoanAndRelatedRecords? I feel like any domain specific concept does not belong here.  Could this method be made more generic?

Reply:
We can. That is what I used for fetching based upon the request related records. 

I didn't change it here because this fix is a high priority change and I wanted to affect the existing code as little as possible and keep the change as small as possible.